### PR TITLE
Modified backward comaptibility rebranding

### PR DESCRIPTION
**Pull request by grount**
It is no longer required that each product will implement the rebranding to retain backward compatibility after package name change.

https://issues.jenkins-ci.org/browse/JENKINS-52352

 
Please Make sure these boxes are checked before submitting your pull request - Thanks ahead!
- [x] Proper pull request title - concise and clear for others.
- [x] Proper pull request short description - clear and concise (as it should appear in the Jira ticket) for other developers and users.
- [x] Proper Jira ticket - Number, Link in pull request description.
- [x] The PR can is merged on your machine without any conflicts.
- [x] The PR can is built on your machine without any (new) warnings.
- [x] The PR passed sanity tests by you / QA / DevTest / Good Samaritan.
- [ ] Add unit tests with new features.
- [ ] If you added any dependency to the POM - Please update grount

### CODE REVIEW

File: src/main/java/com/hpe/application/automation/tools/common/CompatibilityRebrander.java [hunk 1]
```diff
@@ -16,14 +24,34 @@
...
+    public static void addAliasesToAllClasses() {
+        LOG.info("Adding alias for in old package names to add backward compatibility");
+        try {
+            Class[] classes = getClasses(PACKAGE_NAME);
+            for (Class c: classes) {
+                addAliases(c);
+            }
+        }
+        catch(ClassNotFoundException | IOException e) {
+            LOG.warning(e.getMessage());
```
Comment (YafimK) [inline src/main/java/com/hpe/application/automation/tools/common/CompatibilityRebrander.java]: Better use error.
Additional in which cases is the exception is fatal? is there any such case?

File: src/main/java/com/hpe/application/automation/tools/common/CompatibilityRebrander.java [hunk 2]
```diff
@@ -16,14 +24,34 @@
...
+    /**
+     * This function hooks to the Jenkins milestone when the plugins are prepared to load
+     * And adds aliases for all package found classes
+     */
+    @Initializer(before = InitMilestone.PLUGINS_PREPARED)
+    public static void addAliasesToAllClasses() {
+        LOG.info("Adding alias for in old package names to add backward compatibility");
+        try {
+            Class[] classes = getClasses(PACKAGE_NAME);
+            for (Class c: classes) {
```
Comment (YafimK) [inline src/main/java/com/hpe/application/automation/tools/common/CompatibilityRebrander.java]: For debug mode - maybe we should add log.debug for pacakgae name

File: src/main/java/com/hpe/application/automation/tools/common/CompatibilityRebrander.java [hunk 3]
```diff
@@ -36,39 +64,101 @@
...
+     * @throws IOException
      */
-    private static void handleReceivedWrongParameters(@Nonnull Class newClass, String oldClassName, String beforeBrand) {
-        if (!oldClassName.contains(beforeBrand))
-            LOG.warning(String.format("The %s class name doesn't contain: %s class name", newClass.toString(), beforeBrand));
+    private static Class[] getClasses(String packageName) throws ClassNotFoundException, IOException {
+        ClassLoader classLoader = getContextClassLoader();
+        String path = packageName.replace('.', '/');
+        Enumeration resources = classLoader.getResources(path);
+        List directories = getDirectories(resources);
```
Comment (YafimK) [inline src/main/java/com/hpe/application/automation/tools/common/CompatibilityRebrander.java]: Check if you might be able to sue built functions to find the resources

File: src/main/java/com/hpe/application/automation/tools/common/CompatibilityRebrander.java [hunk 4]
```diff
@@ -36,39 +64,101 @@
...
+    }
+
+    /**
+     * The context that is supplied was Jenkins and the project itself, it required the following workaround.
+     * @return ClassLoader that is related to the project context
+     */
+    private static ClassLoader getContextClassLoader() {
+        Thread thread = Thread.currentThread();
+        ClassLoader tempClassLoader = thread.getContextClassLoader();
+        assert tempClassLoader != null;
```
Comment (YafimK) [inline src/main/java/com/hpe/application/automation/tools/common/CompatibilityRebrander.java]: use a logged event so you can trace this afterwards

File: src/main/java/com/hpe/application/automation/tools/common/CompatibilityRebrander.java [hunk 5]
```diff
@@ -36,39 +64,101 @@
...
+        Enumeration resources = classLoader.getResources(path);
+        List directories = getDirectories(resources);
+        ArrayList classes = getClassesArrayList(directories, packageName);
+
+        return (Class[]) classes.toArray(new Class[classes.size()]);
     }
-}
\ No newline at end of file
+
+    private static ArrayList getClassesArrayList(List directories, String packageName) throws ClassNotFoundException, IOException {
```
Comment (YafimK) [inline src/main/java/com/hpe/application/automation/tools/common/CompatibilityRebrander.java]: add docs

**Comment by grount** [comment]
Approve for test build

**Comment by grount** [comment]
Approved for test build

**Pull request merged**